src/picai_prep/dcm2mha.py

```diff
@@ -26,372 +29,389 @@
+        for path in paths:
+            full_path = input_dir / path
+            serie = Series(full_path, patient_id, study_id)
+            try:
+                if not full_path.exists():
+                    raise ArchiveItemPathNotFoundError(path)
+                elif not full_path.is_dir():
+                    raise NotADirectoryError(path)
+                elif path in fullpaths:
+                    raise FileExistsError(path)
+            except Exception as e:
+                serie.error = e
+                logging.error(str(e))
+            else:
+                fullpaths.add(full_path)
+            finally:
+                self.log.append(f'\t+ ({len(self.series)}) {full_path}')
+                self.series.append(serie)
+        if not all([serie.is_valid for serie in self.series]):
+            self.invalidate()
+
+    @property
+    def valid_series(self):
+        return [item for item in self.series if item.is_valid]
+
+    def invalidate(self):
+        for serie in self.valid_series:
+            serie.error = SeriesException('Invalidated due to critical error in sibling')
+
+    def write_log(self, msg: str):
+        self.log.append(msg)
+
+    def compile_log(self):
+        divider = '=' * 120
+        log = [divider,
+               f'CASE {self.patient_id}_{self.study_id}',
+               f'\tPATIENT ID\t{self.patient_id}',
+               f'\tSTUDY ID\t{self.study_id}\n']
+        log += self.log
+        log += ['\nSERIES']
+        for i, serie in enumerate(self.series):
+            log.append(f'({i}) {serie.path.as_posix()}')
+            log.extend([f'\t{l}' for l in serie.log])
+            log.append(f'\tFATAL: {serie.error}\n' if serie.error else '')
+        return '\n'.join(log)
+
+    def convert(self, *args):
+        output_dir, = args
+        self.extract_metadata()
+        self.apply_mappings()
+        if not self.settings.allow_duplicates:
+            self.resolve_duplicates()
+        self.process_and_write(output_dir)
+        return self.compile_log()
+
+    def extract_metadata(self):
+        vseries = len(self.valid_series)
+        self.log.append(f'Extracting metadata from {plural(vseries, "serie")}')
+        errors = []
+
+        file_reader, series_reader = make_sitk_readers()
+        for i, serie in enumerate(self.valid_series):
+            try:
+                dicom_filenames = [os.path.basename(dcm) for dcm in series_reader.GetGDCMSeriesFileNames(serie.path.as_posix())]
+
+                # verify DICOM files are found
+                if len(dicom_filenames) == 0:
+                    raise MissingDICOMFilesError(serie.path)
+
+                if self.settings.verify_dicom_filenames:
+                    serie.verify_dicom_filenames(dicom_filenames)
+
+                serie.extract_metadata(file_reader, dicom_filenames, self.settings.tags)
+                serie.write_log('Extracted metadata')
+            except (MissingDICOMFilesError, UnreadableDICOMError) as e:
```

Comment:
You can use SeriesException to catch both (but also the others)

Reply:
I like being as specific as possible.